Command Flags + Bugs + repr in guild and user

AmashiM opened the pull request.

AmashiM committed:
Fixed message.guild bug

it would cause an error before when trying to get the guild property.

error:
https://media.discordapp.net/attachments/935701677552574506/942563089948233798/unknown.png?width=1440&height=363

AmashiM committed:
fixed repr in guild and user

AmashiM committed:
Merge branch 'VincentRPS:master' into master

AmashiM committed:
Text Command Argument Flags

also some minor errors

VincentRPS commented on discord/ext/commands/core.py:
```diff
@@ -34,11 +35,54 @@
 from .context import Context
 
 
+__all__ = (
```
it would be good if this was a list instead

VincentRPS commented on discord/ext/commands/core.py:
```diff
@@ -34,11 +35,54 @@
+
+
+
+class CommandParser:
```
maybe name this FlagParser instead?

VincentRPS commented on examples/command_flags.py:
```diff
@@ -0,0 +1,30 @@
+
+@bot.command()
+async def ping(ctx):
+    await ctx.send('pong!')
```
maybe add the latency here?

```suggestion
    await ctx.send(f'pong! {round(bot.latency * 1000)}ms')
```

AmashiM replied:
i honestly just copied the command handler script as the base to start implementing but sure thing

VincentRPS reviewed (changes requested):
also unless i did a shit revert it seems like the `__all__`'s are tuples again? maybe change that.

VincentRPS committed:
Merge branch 'master' into master

AmashiM committed:
fixed those things you mentioned in the review

i believe i changed all the __all__ to a list
changed CommandParser to FlagParser
changed the ping command in the examples.

VincentRPS committed:
Update examples/command_flags.py

VincentRPS committed:
Merge branch 'master' into master

AmashiM committed:
Merge branch 'master' of https://github.com/AmashiSenpai/discord.io

AmashiM committed:
Merge branch 'master' of https://github.com/AmashiSenpai/discord.io
